sbi/utils/posterior_ensemble.py

```diff
@@ -0,0 +1,196 @@
+import torch
+from typing import List, Optional
+from torch import Tensor
+from sbi.types import Shape
+
+
+class NeuralPosteriorEnsemble:
+    r"""Wrapper class to bundle together different posterior instances into an ensemble.
+
+    This class creates a posterior ensemble from a set of N different, already trained
+    posterior estimators $p_{i}(\theta|x_o)$, where $i \in \{i,...,N\}$.
+
+    It can wrap all posterior classes available in sbi and even a mixture of different
+    posteriors, i.e. obtained via SNLE and SNPE at the same time, since it only
+    provides a pass-through to the class-methods of each posterior in the ensemble.
+
+    So far `log_prob()`, `sample()` and `map()` functionality are supported.
+
+    Example:
+
+    ```
+    import torch
+    from joblib import Parallel, delayed
+    from sbi.examples.minimal import simple
+
+    ensemble_size = 10
+    posteriors = Parallel(n_jobs=-1)(delayed(simple)() for i in range(ensemble_size))
+
+    ensemble = NeuralPosteriorEnsemble(posteriors)
+    ensemble.set_default_x(torch.zeros((3,)))
+    ensemble.sample((1,))
+    ```
+
+    Args:
+        posteriors: List containing the trained posterior instances that will make up
+        the ensemble.
+        weights: Assign weights to posteriors manually, otherwise they will be
+        weighted with 1/N.
+
+    Attributes:
+        posteriors: List of the posterior estimators making up the ensemble.
+        num_components: Number of posterior estimators.
+        weights: Weight of each posterior distribution. If none are provided each
+            posterior is weighted with 1/N.
+    """
+
+    def __init__(self, posteriors: List, weights: Optional[List] = None):
+        self.posteriors = posteriors
+        self.num_components = len(posteriors)
+        self.weights = weights
+
+    @property
+    def weights(self):
+        return self._weights
+
+    @weights.setter
+    def weights(self, weights: List[float]):
+        """Set relative weight for each posterior in the ensemble.
+
+        Weights are normalised.
+
+        Args:
+            weights: Assignes weight to each posterior distribution.
+        """
+        if type(weights) == list:
+            weights = torch.tensor(weights) / sum(weights)
+            self._weights = weights
+        else:
```

Comment:
I find this weird. If the user passes weights as a torch tensor, the weights won't be used. I suggest to support lists and tensors.